File: 1.1/src/cz/mg/vulkan/PFNvkWaitForFences.java
package cz.mg.vulkan;

public class PFNvkWaitForFences extends VkFunctionPointer {
    public PFNvkWaitForFences() {
    }

    protected PFNvkWaitForFences(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkWaitForFences(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkWaitForFences(long value) {
        setValue(value);
    }

    public PFNvkWaitForFences(VkInstance instance) {
        super(instance, new VkString("vkWaitForFences"));
    }

    public int call(VkDevice device, int fenceCount, VkFence pFences, int waitAll, long timeout){
        return callNative(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), fenceCount, pFences != null ? pFences.getVkAddress() : VkPointer.NULL, waitAll, timeout);
    }

    protected static native int callNative(long vkaddress, long device, int fenceCount, long pFences, int waitAll, long timeout);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkWaitForFences.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkWaitForFences_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jint fenceCount, jlong pFences, jint waitAll, jlong timeout){
    (void)env;
    (void)jc;
    PFN_vkWaitForFences f = (PFN_vkWaitForFences)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        (uint32_t)fenceCount,
        ((VkFence*)jniLongToPointer(pFences)),
        (VkBool32)waitAll,
        (uint64_t)timeout
    );
}

